query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -155,11 +155,16 @@ class SumOverTimeChunkedFunctionD extends SumOverTimeChunkedFunction() with Chun
                                 doubleReader: bv.DoubleVectorDataReader,
                                 startRowNum: Int,
                                 endRowNum: Int): Unit = {
-    // NaN values are ignored by default in the sum method
-    if (sum.isNaN) {
-      sum = 0d
+    val it = doubleReader.iterate(doubleVectAcc, doubleVect, startRowNum)
```

Comment:
Removing doubleReader.sum and iterating will make it slower. Sum function is optimized based on type of data like for DeltaDeltaConstDataReader, DeltaDeltaIterator.
You can do something like:
```
val sumVector = doubleReader.sum(doubleVectAcc, doubleVect, startRowNum, endRowNum)
 if (!sumVector.isNaN && sum.isNaN)
   sum = 0d
 sum += sumVector
```

Reply:
Correct. 

But above snippet will work correctly only if `sum` value inside `doubleReader.sum` is initialized with `Double.NaN` as `sum` both `doubleReader.sum` and `SumOverTimeChunkedFunctionD` is initialized with 0. Not sure, if it is set to 0d for some reason or not on both the places.

So, for PR thought to at least present the idea/bug with the changes done. I also think to make all the required changes in `doubleReader.sum` only.

Reply:
Yes as discussed initial sum should be NaN in all implementations of DoubleVectorDataReader.sum. We cannot make all changes in doubleReader.sum as we need to store sum of all chunks in SumOverTimeChunkedFunction. A window can have multiple chunks, so addTimeLongChunks can be called multiple times. We need to update SumOverTimeChunkedFunction sum for every chunk

Reply:
As suggested, please move this fix into DoubleReader.sum itself -- the variant that reads from double vectors. (`DoubleVectorDataReader`)  This way if the vector is integral, we use the faster sum which doesn't need to test for NaN.